From d8924cc1a4a36b6a9debfa074bb3b36d3800a42a Mon Sep 17 00:00:00 2001
From: Daniel Goehring <dgoehrin@os.amperecomputing.com>
Date: Fri, 7 May 2021 13:53:34 -0400
Subject: [PATCH 29/36] ac03: Silicon user config files [DO NOT UPSTREAM]

Add user configuration files for AmpereOne silicon

Configured for the Olimex ARM-USB-OCD_H JTAG probe.
https://www.olimex.com/Products/ARM/JTAG/ARM-USB-OCD-H/

OpenOCD configured for the following AmpereOne options
1. One and two socket platforms

Change-Id: I196af0a39cb920427d1f489fa2699a5e97edb704
Signed-off-by: Daniel Goehring <dgoehrin@os.amperecomputing.com>
---
 tcl/openocd.ac03_1s_sil.cfg | 89 +++++++++++++++++++++++++++++++++
 tcl/openocd.ac03_2s_sil.cfg | 98 +++++++++++++++++++++++++++++++++++++
 2 files changed, 187 insertions(+)
 create mode 100644 tcl/openocd.ac03_1s_sil.cfg
 create mode 100644 tcl/openocd.ac03_2s_sil.cfg

diff --git a/tcl/openocd.ac03_1s_sil.cfg b/tcl/openocd.ac03_1s_sil.cfg
new file mode 100644
index 000000000..192f17706
--- /dev/null
+++ b/tcl/openocd.ac03_1s_sil.cfg
@@ -0,0 +1,89 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Example OpenOCD user configuration for the AmpereOne
+# Development Platform
+#
+# Copyright (c) 2021-2022, Ampere Computing LLC
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "4000" kHz
+# Syntax: -c "set JTAGFREQ {4000}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0
+# Specify available physical cores by number
+# Example syntax to attach to physical cores 16, 17, 94, 95, 128 and 129 for S0
+# Syntax: -c "set CORELIST_S0 {16 17 94 95 128 129}"
+#
+# COREMASK_S0_0
+# Specify available physical cores 0-63 by mask
+# Example syntax to attach to physical cores 16 and 17 for S0
+# Syntax: -c "set COREMASK_S0_0 {0x0000000000030000}"
+#
+# COREMASK_S0_1
+# Specify available physical cores 64-127 by mask
+# Example syntax to attach to physical cores 94 and 95 for S0
+# Syntax: -c "set COREMASK_S0_1 {0x00000000C0000000}"
+#
+# COREMASK_S0_2
+# Specify available physical cores 128 and above by mask
+# Example syntax to attach to physical cores 128 and 129 for S0
+# Syntax: -c "set COREMASK_S0_2 {0x0000000000000003}"
+
+# Operating System Environment Variable Overrides
+#
+# OCD_GDB_PORT
+# Specify OpenOCD's GDB port
+# csh:  setenv OCD_GDB_PORT 3333
+# bash: export OCD_GDB_PORT=3333
+#
+# OCD_TELNET_PORT
+# Specify OpenOCD's Telnet port
+# csh:  setenv OCD_TELNET_PORT 4444
+# bash: export OCD_TELNET_PORT=4444
+#
+# OCD_TCL_PORT
+# Specify OpenOCD's TCL port
+# csh:  setenv OCD_TCL_PORT 6666
+# bash: export OCD_TCL_PORT=6666
+
+bindto 0.0.0.0
+#debug_level 3
+
+if { [info exists env(OCD_GDB_PORT)] } {
+	gdb_port $::env(OCD_GDB_PORT)
+}
+
+if { [info exists env(OCD_TELNET_PORT)] } {
+	telnet_port $::env(OCD_TELNET_PORT)
+}
+
+if { [info exists env(OCD_TCL_PORT)] } {
+	tcl_port $::env(OCD_TCL_PORT)
+}
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 4000
+}
+
+source [find board/ampere_ac03_1s.cfg]
diff --git a/tcl/openocd.ac03_2s_sil.cfg b/tcl/openocd.ac03_2s_sil.cfg
new file mode 100644
index 000000000..575e7850d
--- /dev/null
+++ b/tcl/openocd.ac03_2s_sil.cfg
@@ -0,0 +1,98 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Example OpenOCD user configuration for the AmpereOne
+# Development Platform
+#
+# Copyright (c) 2021-2022, Ampere Computing LLC
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "4000" kHz
+# Syntax: -c "set JTAGFREQ {4000}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0, CORELIST_S1
+# Specify available physical cores by number
+# Example syntax to attach to physical cores 16, 17, 94, 95, 128 and 129 for S0 and S1
+# Syntax: -c "set CORELIST_S0 {16 17 94 95 128 129}"
+# Syntax: -c "set CORELIST_S1 {16 17 94 95 128 129}"
+#
+# COREMASK_S0_0, COREMASK_S1_0
+# Specify available physical cores 0-63 by mask
+# Example syntax to attach to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set COREMASK_S0_0 {0x0000000000030000}"
+# Syntax: -c "set COREMASK_S1_0 {0x0000000000030000}"
+#
+# COREMASK_S0_1, COREMASK_S1_1
+# Specify available physical cores 64-127 by mask
+# Example syntax to attach to physical cores 94 and 95 for S0 and S1
+# Syntax: -c "set COREMASK_S0_1 {0x00000000C0000000}"
+# Syntax: -c "set COREMASK_S1_1 {0x00000000C0000000}"
+#
+# COREMASK_S0_2, COREMASK_S1_2
+# Specify available physical cores 128 and above by mask
+# Example syntax to attach to physical cores 128 and 129 for S0 and S1
+# Syntax: -c "set COREMASK_S0_2 {0x0000000000000003}"
+# Syntax: -c "set COREMASK_S1_2 {0x0000000000000003}"
+#
+# SPLITSMP
+# Group all ARMv8 cores per socket into individual SMP sessions
+# If not specified, group ARMv8 cores from both sockets into one SMP session
+# Syntax: -c "set SPLITSMP {}"
+
+# Operating System Environment Variable Overrides
+#
+# OCD_GDB_PORT
+# Specify OpenOCD's GDB port
+# csh:  setenv OCD_GDB_PORT 3333
+# bash: export OCD_GDB_PORT=3333
+#
+# OCD_TELNET_PORT
+# Specify OpenOCD's Telnet port
+# csh:  setenv OCD_TELNET_PORT 4444
+# bash: export OCD_TELNET_PORT=4444
+#
+# OCD_TCL_PORT
+# Specify OpenOCD's TCL port
+# csh:  setenv OCD_TCL_PORT 6666
+# bash: export OCD_TCL_PORT=6666
+
+bindto 0.0.0.0
+#debug_level 3
+
+if { [info exists env(OCD_GDB_PORT)] } {
+	gdb_port $::env(OCD_GDB_PORT)
+}
+
+if { [info exists env(OCD_TELNET_PORT)] } {
+	telnet_port $::env(OCD_TELNET_PORT)
+}
+
+if { [info exists env(OCD_TCL_PORT)] } {
+	tcl_port $::env(OCD_TCL_PORT)
+}
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 4000
+}
+
+source [find board/ampere_ac03_2s.cfg]
-- 
2.25.1

